pagesections/ComponentInteraction.js

```diff
@@ -1,47 +1,63 @@
 'use strict';
 
-const ComponentInteraction = ( Base ) => class extends Base {
-
-	static get OOUI_SELECTORS() {
-		const visibleCondition = 'not(contains(@style,"display: none"))';
-
-		return {
-			LOOKUP_OPTION_WIDGET: '.oo-ui-lookupElement-menu .oo-ui-optionWidget',
-			MULTI_OPTION_WIDGET: '.oo-ui-optionWidget',
-			OPTION_WIDGET_SELECTED: '.oo-ui-optionWidget-selected',
-			OVERLAY: '.oo-ui-defaultOverlay',
-			COMBOBOX_DROPDOWN: '.oo-ui-comboBoxInputWidget-dropdownButton',
-			VISIBLE_ENTITY_SUGGESTION: `//ul[contains(@class, "ui-suggester-list")][${visibleCondition}]//li`
-		};
-	}
-
-	setValueOnLookupElement( element, value ) {
-		element.$( 'input' ).setValue( value );
-		$( 'body' ).$( this.constructor.OOUI_SELECTORS.LOOKUP_OPTION_WIDGET ).waitForDisplayed();
-		$( this.constructor.OOUI_SELECTORS.LOOKUP_OPTION_WIDGET ).click();
-	}
-
-	setSingleValueOnMultiselectElement( element, value ) {
-		element.$( 'input' ).setValue( value );
-		element.$( this.constructor.OOUI_SELECTORS.MULTI_OPTION_WIDGET ).waitForDisplayed();
-		element.$( this.constructor.OOUI_SELECTORS.MULTI_OPTION_WIDGET ).click();
-	}
-
-	setValueOnComboboxElement( element, value ) {
-		element.$( 'input' ).setValue( value );
-		browser.waitUntil( () => {
-			return browser.isDisplayed(
-				`${this.constructor.OOUI_SELECTORS.OVERLAY} ${this.constructor.OOUI_SELECTORS.OPTION_WIDGET_SELECTED}`
-			);
-		} );
-		// close suggestion overlay
-		element.$( this.constructor.OOUI_SELECTORS.COMBOBOX_DROPDOWN ).click();
-	}
-
-	async selectFirstSuggestedEntityOnEntitySelector() {
-		await $( this.constructor.OOUI_SELECTORS.VISIBLE_ENTITY_SUGGESTION ).waitForDisplayed();
-		await $( this.constructor.OOUI_SELECTORS.VISIBLE_ENTITY_SUGGESTION ).click();
-	}
+const ComponentInteraction = ( Base ) => {
+	/** @extends Base */
+	return class ComponentInteractionMixin extends Base {
```

Comment:
Hmm, what's the reason for this change? `/** @extends Base */` is pretty redundant to begin with since that's what the line below says, and since `Base` is dynamic, there are no type hints to be unlocked anyway. If there is no benefit I'm missing, I'd prefer getting rid of this one and going back to the shorthand return syntax to remove the extra indentation level.

Same in the two other cases below.

Reply:
Good question! It does indeed look to be redundant, but I found that it was required in order for the IDE to realize that `ItemPage` has methods from the "pagesections" classes (e.g. `ItemPage.selectFirstSuggestedEntityOnEntitySelector()`). If you can think of a better way to hint that I'm all ears.

Reply:
but... why...